From f1d9563ec93ddd5f9e00c7291e1be82bbcbd6e19 Mon Sep 17 00:00:00 2001
Message-Id: <f1d9563ec93ddd5f9e00c7291e1be82bbcbd6e19.1467309786.git.jen@redhat.com>
In-Reply-To: <b75cb6f28f374e6cee8f45cb30d017d2cf77dcf6.1467309786.git.jen@redhat.com>
References: <b75cb6f28f374e6cee8f45cb30d017d2cf77dcf6.1467309786.git.jen@redhat.com>
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 14 Jun 2016 08:53:45 -0500
Subject: [CHANGE 08/10] vga: update vga register setup on vbe changes
To: rhvirt-patches@redhat.com,
    jen@redhat.com

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1465894427-26109-9-git-send-email-kraxel@redhat.com>
Patchwork-id: 70597
O-Subject: [RHEL-6.9 qemu-kvm PATCH v3 08/10] vga: update vga register setup on vbe changes
Bugzilla: 1331408
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

Call the new vbe_update_vgaregs() function on vbe configuration
changes, to make sure vga registers are up-to-date.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(Upstream commit ID: 2068192dcccd8a80dddfcc8df6164cf9c26e0fc4)
---
 hw/vga.c | 1 +
 1 file changed, 1 insertion(+)

Signed-off-by: Jeff E. Nelson <jen@redhat.com>
---
 hw/vga.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/vga.c b/hw/vga.c
index a340e89..7e812a4 100644
--- a/hw/vga.c
+++ b/hw/vga.c
@@ -726,6 +726,7 @@ static void vbe_ioport_write_data(void *opaque, uint32_t addr, uint32_t val)
         case VBE_DISPI_INDEX_Y_OFFSET:
             s->vbe_regs[s->vbe_index] = val;
             vbe_fixup_regs(s);
+            vbe_update_vgaregs(s);
             break;
         case VBE_DISPI_INDEX_BANK:
             val &= s->vbe_bank_mask;
-- 
2.5.5

